src/bd/primitives/Specializer.java

```diff
@@ -0,0 +1,118 @@
+package bd.primitives;
+
+import java.lang.reflect.InvocationTargetException;
+
+import com.oracle.truffle.api.dsl.NodeFactory;
+import com.oracle.truffle.api.source.SourceSection;
+
+import bd.primitives.Primitive.NoChild;
+import bd.settings.VmSettings;
+
+
+/**
+ * A Specializer defines when a node can be used as a eager primitive, how
+ * it is to be instantiated, and acts as factory for them.
+ *
+ * @param <T> the type of the nodes created by this specializer
+ * @param <Context> the type of the context object
+ * @param <ExprT> the root type of expressions used by the language
+ */
+public class Specializer<T, Context, ExprT> {
+  protected final Context                    context;
```

Comment:
Grace's spec uses neither of these terms, so I don't have any opinion in so far as how I plan to use these. I suppose I'd prefer `frame` out of the two (since it feels more general to me).